flytekit/annotated/workflow.py

```diff
@@ -32,6 +35,49 @@
 )
 
 
+@dataclass
+class WorkflowMetadata(object):
+    class OnFailurePolicy(Enum):
+        FAIL_IMMEDIATELY = _workflow_model.WorkflowMetadata.OnFailurePolicy.FAIL_IMMEDIATELY
+        FAIL_AFTER_EXECUTABLE_NODES_COMPLETE = (
+            _workflow_model.WorkflowMetadata.OnFailurePolicy.FAIL_AFTER_EXECUTABLE_NODES_COMPLETE
+        )
+
+    on_failure: OnFailurePolicy
+
+    def __post_init__(self):
+        if (
+            self.on_failure != self.OnFailurePolicy.FAIL_IMMEDIATELY
+            and self.on_failure != self.OnFailurePolicy.FAIL_AFTER_EXECUTABLE_NODES_COMPLETE
+        ):
+            raise FlyteValidationException(f"Failure policy {self.on_failure} not acceptable")
+
+    def to_flyte_model(self):
+        if self.on_failure == WorkflowMetadata.OnFailurePolicy.FAIL_IMMEDIATELY:
+            on_failure = 0
+        else:
+            on_failure = 1
+        return _workflow_model.WorkflowMetadata(on_failure=on_failure)
+
+
+@dataclass
+class WorkflowMetadataDefaults(object):
```

Comment:
can we call this something else like `WorkflowDefaults` much better than WorkflowMetadataDefaults
And should this be part of WorkflowMetadata?

Reply:
I'm okay with this as it stands but up to you.  They are two separate objects in the IDL so I just kept that pattern, it makes the to_flyte_model function easier to reason about.  Also, these aren't exposed, only the settings themselves are in the decorator.